projects/pgai/pgai/vectorizer/vectorizer.py

```diff
@@ -464,13 +467,177 @@ async def run(self) -> int:
             await register_vector_async(conn)
             while True:
                 if not self._continue_processing(loops, res):
+                    await self._check_and_process_openai_batches(conn)
+                    # TODO how can we run this only after hitting the rate limit of the normal openai batch embedding api?
+                    await self._do_openai_batch(conn)
                     return res
                 items_processed = await self._do_batch(conn)
                 if items_processed == 0:
                     return res
                 res += items_processed
                 loops += 1
 
+    async def _create_batch_table(self, conn: AsyncConnection):
+        # TODO this does not feel like the way to go, is there a way to do these kind of migrations properly?
```

Comment:
Have a look at the extension code and the create_vectorizer function in particular. We create e.g. the queue table there so this code should probably also live in there.
I also think we need one of these for each vectorizer and not a shared one for all.

Reply:
I imagine I'll need to create a migration for this? Do I need to redefine everything or only the `create_vectorizer` function?